packages/skeleton-svelte/src/lib/components/ProgressRing/ProgressRing.svelte

```diff
@@ -0,0 +1,106 @@
+<script lang="ts">
+	import type { ProgressRingProps } from './types.js';
+
+	let {
+		value,
+		max = 100,
+		strokeWidth = 50, // px
+		strokeLinecap = 'round',
+		labelledby = '',
+		// Base (figure)
+		base = 'relative',
+		size = 'size-32',
+		classes = '',
+		// Slot
+		slotBase = 'absolute top-0 left-0 z-[1] flex justify-center items-center',
+		slotClasses = '',
+		// SVG
+		svgBase = 'absolute top-0 left-0 w-full h-full rounded-full',
+		svgClasses = '',
+		// Track
+		trackBase = 'fill-none',
+		trackStroke = 'stroke-surface-200-800',
+		trackClasses = '',
+		// Meter
+		meterBase = 'fill-none',
+		meterStroke = 'stroke-primary-500',
+		meterTransition = 'transition-[stroke-dashoffset]',
+		meterDuration = 'duration-100',
+		meterClasses = '',
+		// Label
+		label,
+		labelBase = '',
+		labelFill = 'fill-surface-950-50',
+		labelFontSize = 96, // px
+		labelFontWeight = 'bold',
+		labelClasses = '',
+		// Snippets
+		children
+	}: ProgressRingProps = $props();
+
+	// Local
+	const baseSize = 512; // px
+	const radius: number = baseSize / 2 - strokeWidth / 2;
+	let circumference: number = $state(radius);
+	let dashoffset: number = $state(calcDashOffset());
+
+	$effect(() => {
+		dashoffset = calcDashOffset();
+	});
+
+	function calcDashOffset() {
+		const _value = value !== undefined ? value : 25;
```

Comment:
I think "valueFallback" is better for the code readability here.

Reply:
It's not always a fallback though, it's either the value or fallback. But I'll think on it.

Reply:
I'll just keep it simple for now and go with `v`. It's a local variable and not user-facing, so it should be fine.